packages/core/src/alert/alert-actions.element.ts

```diff
@@ -78,35 +26,40 @@ function updateLayoutsAndActionButtonStyles(alertActionsComponent: CdsAlertActio
  * </cds-alert>
  * ```
  *
- * @beta
  * @element cds-alert-actions
- * @slot default
+ * @slot
  * @cssprop --action-text-color
  * @cssprop --action-hover-text-color
  * @cssprop --action-font-size
  */
 export class CdsAlertActions extends LitElement {
   /**
    * @type {default | banner | light}
+   * @private
    */
   @property({ type: String })
   type: AlertGroupTypes = 'light';
 
-  /** @private */
-  @querySlotAll('cds-button') buttons: NodeListOf<CdsButton>;
+  @querySlotAll('cds-button')
+  private buttons: NodeListOf<CdsButton>;
+
+  render() {
+    return html`<div class="private-host">
+      <slot cds-layout="horizontal wrap:none gap:sm align:vertical-center"></slot>
+    </div>`;
+  }
 
   connectedCallback() {
     super.connectedCallback();
     assignSlotNames([this, 'actions']);
-    updateLayoutsAndActionButtonStyles(this, this.parentElement as CdsAlert);
   }
 
-  updated() {
-    updateLayoutsAndActionButtonStyles(this, this.parentElement as CdsAlert);
```

Comment:
The main source of the lifecycle issues is when the child component refers to the parent. Depending on how the framework renders and inserts the element into the DOM it could be inversed order which breaks the lookup. This was happening in vuepress and angular. I removed a similar parentElement reference in forms.

For the cases of needing to pass data to the child we can use the `@internalProperty` decorator which will work regardless of render order. It will also mark it as an internal API omitting it from the docs.